### ISSUE TITLE
[Feature] Generalized support for external IDs

### ISSUE THREAD

**Issue by stg-annon** [stashapp/stash#1790]
cross posting this from [stash-box](https://github.com/stashapp/stash-box/issues/113) to track progress in Stash

**Is your feature request related to a problem? Please describe.**
Stash currently only supports the ability to store external ids from stash-boxes, this feature request is to allow for the `stash_id` concept to be generalized to allow for other external sources of metadata

**Describe the solution you'd like**
the `stash_id` tables in stash to allow for alternative links or an alternative table be created e.g. 

if a user wants to use TPDB for metadata an entry into either table it could be entered like the following
| {item}_id | endpoint | external_id |
|---|---|---|
| 1234 | api.metadataapi.net/{item}/ | a1991648-22ce-11ec-9621-0242ac130002 |

This concept could also be extended to other external links, like links to the media source, social links, etc. like described in in [stash-box#113](https://github.com/stashapp/stash-box/issues/113)

**Describe alternatives you've considered**
Storing this external match data in a separate DB, this is not ideal because it does not integrate with the stash API or stash UI

**Comment by laurus-lx** [comment]
FYI - while we are waiting for stash to officially support this, there is a way to add performer handles as is - open stash db in your favorite admin  tool (can recommend DBeaver) - add URLs to stash_id table and they'll show up in performer page under stash_id links
![image](https://user-images.githubusercontent.com/83609083/135898850-c020720e-88fe-4b68-9758-d41759c50668.png)

https://discord.com/channels/559159668438728723/559399634318131210

**Comment by stg-annon** [comment]
> 
> 
> I think generalising (and renaming) the `_stash_ids` tables is a good idea and will allow better integration with the scraper system. I do however think that it should be separated from a generalised external link table.

I agree, They are similar but different things and should be handled differently, would the handling of links be better suited in an entirely different issue?

**Comment by GalacticCheat** [comment]
Maista sucks cock
![Image](https://github.com/user-attachments/assets/ed12369e-cb07-434a-b49f-fbc470272035)
https://i.ibb.co/ds49xcXt/image-2-copie.jpg
